### PR TITLE
Added unique identifiers for all actions, issue #619

### PR DESCRIPTION
**Pull request by joacimb**
Fixes the error message when opening the toolbar customization tool after the locale has been switched. Issue #619

The identifiers unfortunately needs the addition of a magic string. Decide if you think this is the right solution. :)

This could maybe be solved by using the action class name. But this requires that the action only has one usage, as opposed to OverrideAction for example which has many actions built into it.

### CODE REVIEW

**Comment by winder** [comment]
@joacimb wow, thanks for digging into this. The programmatic way I wanted to handle localization is non-standard for Netbeans Platform applications... which led to the piece of code you found here.

I'll need to do a little bit of testing, but this looks like a good solution.

File: ugs-core/src/com/willwinder/universalgcodesender/i18n/Localization.java [hunk 1]
```diff
@@ -56,18 +55,20 @@ synchronized public static boolean initialize(String language) {
...
-        locale = new Locale(language, region);
+        Locale locale = new Locale(language, region);
+        Locale.setDefault(locale);
         bundle = ResourceBundle.getBundle("resources.MessagesBundle", locale);
         return getKeyCount(bundle) >= getEnglishKeyCount();
     }
 
     public static String loadedLocale() {
-        return locale + "";
+        return Locale.getDefault() + "";
```
Comment (winder) [inline ugs-core/src/com/willwinder/universalgcodesender/i18n/Localization.java]: This isn't going to have the expected behavior, the `getDefault` call needs to go to the location where `initialize` is being called. I load some GRBL code number translation files with this (even though GRBL doesn't have localizations yet).

Comment (joacimb) [inline ugs-core/src/com/willwinder/universalgcodesender/i18n/Localization.java]: Hmm, I could be missing something, but I thought that by storing the selected locale using "Locale.setDefault" in the initialize-method would be the same as storing it as a local member?

Comment (winder) [inline ugs-core/src/com/willwinder/universalgcodesender/i18n/Localization.java]: Thanks, I understand now, I thought this was changing the UGS localization based on the system locale. I didn't realize the `Locale` defaults worked this way.

**Comment by winder** [comment]
I tried out the Locale changes but it didn't seem to localize the NetBeans menus, are those bundles separate downloads?

**Comment by joacimb** [comment]
I have not found any translation bundles for other languages for the standard platform. I tried overriding translations for existing menu items using this in `layers.xml`. Some menu items can be translated while others (eg. Help) won't for some reason...
```xml
<folder name="File">
    <attr name="SystemFileSystem.localizingBundle" stringvalue="resources.MessagesBundle"/>
    <attr name="displayName" bundlevalue="#file"/>
</folder>
```

**Pull request merged**